definitions.asm

```diff
@@ -2,33 +2,77 @@
 ; Scot W. Stevenson <scot.stevenson@gmail.com>
 ; Modified by Patrick Surry
 ; First version: 01. Apr 2016 (Liara Forth)
-; This version: 21. Apr 2024
+; This version: 1. Jun 2024
 
 ; This file is included by taliforth.asm. These are the general
 ; definitions; platform-specific definitions such as the
 ; memory map are kept in the platform folder.
 
-; TaliForth reserves the first part of zero page ($0 - zpage_end) which is
-; configured by zpage_end in platform/platform-*.asm and normally 128 bytes.
-; The rest of zero page is free for kernel/external use (zpage_end+1 - $FF)
-; TaliForth usage is as follows:
-;   zero page variables: 30 words = 60 bytes ($0000-$0038)
-;   Forth Data Stack: 128 - 60 - 8 = 60 bytes or 30 words
-;   Data Stack floodplain: 8 bytes after stack (to avoid catastrophic underflow)
+; HARD PHYSICAL ADDRESSES
+
+; Some of these are somewhat silly for the 65c02, where for example
+; the location of the Zero Page is fixed by hardware. However, we keep
+; these for easier comparisons with Liara Forth's structure and to
+; help people new to these things.
+
+ram_start = $0000          ; start of installed RAM, must include zpage
+zpage     = ram_start      ; begin of Zero Page ($0000-$00ff)
+stack0    = $0100          ; begin of Return Stack ($0100-$01ff)
+
+; weak constants can be overridden by the including platform configuration
+.weak
+zpage_end = $7F            ; end of Zero Page used ($0000-$007f)
+ram_end   = $8000-1        ; end of installed RAM
+hist_buff = ram_end-$03ff  ; begin of history buffers
+.endweak
+
+; SOFT PHYSICAL ADDRESSES
+
+; Tali currently doesn't have separate user variables for multitasking. To
+; prepare for this, though, we've already named the location of the user's
+; Zero-Page System Variables user0. Note cp0 starts one byte further down so
+; that it currently has the address $300 and not $2FF. This avoids crossing
+; the page boundry when accessing the RAM System Variables table, which would
+; cost an extra cycle.
+
+user0     = zpage            ; TaliForth2 system variables
+rsp0      = $ff              ; initial Return Stack Pointer (65c02 stack)
+bsize     = $ff              ; size of input/output buffers
+
+.weak
```

Comment:
I like what you are doing with weak definitions in definitions.asm that the user can override in their platform file.  I'd include `user0` in the weak definitions, as it's not always at the beginning of zero page (eg when there is already an OS that uses zero page and a few systems that have IO at the bottom of zero page)